compiler/modules/types/list.bal

```diff
@@ -151,31 +180,14 @@ function listFormulaIsEmpty(Context cx, Conjunction? pos, Conjunction? neg) retu
                 Atom d = p.atom;
                 p = p.next; 
                 lt = cx.listAtomType(d);
-                int newLen = int:max(members.length(), lt.members.length());
-                if members.length() < newLen {
-                    if isNever(rest) {
-                        return true;
-                    }
-                    // JBUG #33532 should be able to use `_` here
-                    foreach int i in members.length() ..< newLen {
-                        members.push(rest);
-                    }
-                }
-                foreach int i in 0 ..< lt.members.length() {
-                    members[i] = intersect(members[i], lt.members[i]);
-                }
-                if lt.members.length() < newLen {
-                    if isNever(lt.rest) {
-                        return true;
-                    }
-                    foreach int i in lt.members.length() ..< newLen {
-                        members[i] = intersect(members[i], lt.rest);
-                    }
+                var intersected = listIntersectWith(members, rest, lt);
+                if intersected is () {
+                    return true;
                 }
-                rest = intersect(rest, lt.rest);
+                [members, rest] = intersected;
             }
         }
-        foreach var m in members {
+        foreach var m in members.initial {
```

Comment:
We have this both here and in listProj, so let's factor it into a function typeListAllEmpty in common.bal.

Reply:
typeListAnyEmpty?

Reply:
Since we operate on the level of `FixedLengthArray`, shouldn't this function take a `FixedLengthArray` as an argument, in that case we will have to keep the factored function in `list.bal` I think.

Reply:
That's fine too. And yes, should be any not every.